cloudinit/sources/DataSourceTencentCloud.py

```diff
@@ -0,0 +1,259 @@
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import json
+import os
+import traceback
+from typing import List
+
+from cloudinit import dmi
+from cloudinit import log as logging
+from cloudinit import sources, ssh_util, util
+from cloudinit.sources import DataSourceEc2 as EC2
+from cloudinit.sources import DataSourceHostname
+
+TENCENTCLOUD_PRODUCT = "Tencent Cloud CVM"
+LOG = logging.getLogger(__name__)
+
+
+class DataSourceTencentCloud(EC2.DataSourceEc2):
+
+    metadata_urls = ["http://169.254.0.23", "http://metadata.tencentyun.com"]
+
+    # The minimum supported metadata_version from the ec2 metadata apis
+    min_metadata_version = "2017-09-19"
+    extended_metadata_versions: List[str] = []
+
+    def __init__(self, sys_cfg, distro, paths):
+        super(DataSourceTencentCloud, self).__init__(sys_cfg, distro, paths)
+        self.seed_dir = os.path.join(paths.seed_dir, "TencentCloud")
+
+    def set_config_mod_always(self, mod_name):
+        self.set_mod_always(mod_name, "cloud_config_modules")
+
+    def set_mod_always(self, mod_name, stage):
+        modules = self.sys_cfg.get(stage, [])
+        new_modules = []
+        for mod in modules:
+            if mod == mod_name:
+                new_modules.append([mod_name, "always"])
+            else:
+                new_modules.append(mod)
+        self.cfg[stage] = new_modules
+
+    def add_config_mod_always(self, mod_name):
+        modules = self.sys_cfg.get("cloud_config_modules", [])
+        modules.append([mod_name, "always"])
+        self.cfg["cloud_config_modules"] = modules
+
+    def check_item_change(self, item, value):
+        if not value and item != "ssh_key":
+            return False
+        changed = False
+        d_item = {}
+        item_path = "/var/lib/cloud/tencentcloud.item"
+        try:
+            if os.path.exists(item_path):
+                with open(item_path, "r") as f:
+                    d_item = json.load(f)
+                if d_item.get(item) != value:
+                    changed = True
+            else:
+                changed = True
+            if changed:
+                d_item[item] = value
+                with open(item_path, "w") as f:
+                    json.dump(d_item, f)
+        except Exception as e:
+            LOG.debug(str(e))
+            LOG.debug(traceback.format_exc())
```

Comment:
this logging seems excessive

Reply:
thanks,I remove this line "LOG.debug(traceback.format_exc())"